### ISSUE TITLE
CA1508 should not warn when string.IsNullOrWhiteSpace(string!)

### ISSUE THREAD

**Issue by JohanLarsson** [dotnet/roslyn-analyzers#4653]
```cs
void M(object o)
{
    if (o is string s)
    {
        if (string.IsNullOrWhiteSpace(s)) // s may still be whitespace
        {
        }
    }
}
```

![image](https://user-images.githubusercontent.com/1640096/103445463-a7cbca00-4c74-11eb-9385-09e1ff317376.png)

**Comment by Youssef1313** [comment]
This was already fixed in #4439

Related unit test:

https://github.com/dotnet/roslyn-analyzers/blob/e8d8d621d066bbf199e4f7c34fbedbefc6c9ea2d/src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/Maintainability/AvoidDeadConditionalCode_NullAnalysis.cs#L6887-L6914
